lib/Net/SAML2/Binding/SOAP.pm

```diff
@@ -104,16 +105,40 @@ has 'anchors' => (
     predicate => 'has_anchors'
 );
 
-=head2 request( $message )
+# BUILDARGS
+
+# Earlier versions expected the idp_cert to be a string.  However, metadata
+# can include multiple signing certificates so the $idp->cert is now
+# expected to be an arrayref to the certificates.  To avoid breaking existing
+# applications this changes the the cert to an arrayref if it is not
+# already an array ref.
+
+around BUILDARGS => sub {
+    my $orig = shift;
+    my $self = shift;
+
+    my %params = @_;
+    if ($params{idp_cert} && ref($params{idp_cert}) ne 'ARRAY') {
+            $params{idp_cert} = [$params{idp_cert}];
+    }
+
+    return $self->$orig(%params);
+};
+
+=head2 request( $message, $trust_insecure )
 
 Submit the message to the IdP's service.
 
 Returns the Response, or dies if there was an error.
 
+Optionally if $trust_insecure is true it will not check for a signature.
+Shibboleth in particular may not sign responses that it considers to be
+transfered on a secure channel.
+
```

Comment:
I have an email into the mailing list - I would prefer a config item in Shibboleth but recognise not everyone can control the IdP